src/AuthenticatedAPIClient/authInterface.js

```diff
@@ -81,5 +82,25 @@ export default function applyAuthInterface(authenticatedAPIClient, authConfig) {
 
   authenticatedAPIClient.isCsrfExempt = url =>
     authenticatedAPIClient.csrfExemptUrls.includes(url);
+
+  authenticatedAPIClient.validateCookies = (route = '') => {
+    // Validate auth-related cookies are in a consistent state.
+    const cookies = new Cookies();
+    const hasAccessToken = !!cookies.get(authenticatedAPIClient.accessTokenCookieName);
+    const hasUserInfo = !!cookies.get(authenticatedAPIClient.userInfoCookieName);
+    if (!hasAccessToken) {
+      if (hasUserInfo) {
+        // Cookies are out of sync. Log the user out to reset cookies and resync.
+        logError(new Error(`Invalid auth cookies: hasAccessToken=${hasAccessToken}, hasUserInfo=${hasUserInfo}`));
+        authenticatedAPIClient.logout(authenticatedAPIClient.appBaseUrl + route);
+      } else {
+        authenticatedAPIClient.login(authenticatedAPIClient.appBaseUrl + route);
```

Comment:
I feel like a name like `ensureAuthencationAndCookies` would make me a little less surprised that this may result in logout or login.

It might be cleaner to separate out clean-up from straight authentication, but that would require a little more thought.